src/Fields/broadcast.jl

```diff
@@ -32,15 +32,63 @@ Base.Broadcast.broadcastable(field::Field) = field
 Base.eltype(bc::Base.Broadcast.Broadcasted{<:AbstractFieldStyle}) =
     Base.Broadcast.combine_eltypes(bc.f, bc.args)
 
+# to_struct: for recursively converting a point field into a struct
+to_struct(x::DataLayouts.DataF{T}) where {T <: Geometry.AbstractPoint} =
+    T(parent(x)...)
+to_struct(x::DataLayouts.DataF{T}) where {T <: Real} = T(first(parent(x)))
+to_struct(x::DataLayouts.DataF{T}) where {T <: StaticArrays.SMatrix} =
+    T(parent(x)...)
+to_struct(x::DataLayouts.DataF{T}) where {T <: Geometry.AxisTensor} =
+    T(axes(T), to_struct.(getproperty.(Ref(x), propertynames(x)))...)
+to_struct(lg::DataLayouts.DataF{T}) where {T <: Geometry.LocalGeometry} =
+    T((to_struct.(getproperty.(Ref(lg), propertynames(lg))))...)
+to_struct(x::DataLayouts.DataF{T}) where {T <: NamedTuple} =
+    T(to_struct.(getproperty.(Ref(x), propertynames(x))))
```

Comment:
can't we just call `x[]`, so that you hit this method:
https://github.com/CliMA/ClimaCore.jl/blob/main/src/DataLayouts/DataLayouts.jl#L494-L496

Reply:
Yes, I had to rearrange the dispatching a bit, but I think this works now.